### PR TITLE
Add logging and assertion for update workout

### PR DESCRIPTION
**Pull request by TianaiYan**
Inspired by Alan, update workout should also check whether the workout to update exists in the list.

### CODE REVIEW

File: src/main/java/data/workouts/WorkoutList.java [hunk 1]
```diff
@@ -207,39 +207,53 @@ private boolean checkIndexIsWithinRange(int index) {
...
         }
 
         Workout updatedWorkout = workoutsList.get(indexToUpdate - 1);
-        updatedWorkout.setRepetitions(newNumberOfReps);
+        String exerciseName = updatedWorkout.getExerciseName();
+        boolean isWorkoutExist = checkForExistingWorkout(exerciseName, newRepsValue);
+
+        if (isWorkoutExist) {
+            logger.log(Level.WARNING, "\"" + exerciseName + " (" + newRepsValue + ")\" " +
+                "has already existed in the list. Cannot be updated." );
```
Comment (alanlowzies) [inline src/main/java/data/workouts/WorkoutList.java]: (Reported by Gradle's checkstyle module) The closing parenthesis on Line 250 has an extra space before it.

**Review by alanlowzies — COMMENTED** [review]
The logs and assertions look alright. However, I noticed that the Java CI checks failed. I've included the areas that the checks pointed out for you to take a look. Thanks :)

File: src/main/java/data/workouts/WorkoutList.java [hunk 2]
```diff
@@ -207,39 +207,53 @@ private boolean checkIndexIsWithinRange(int index) {
...
-        if (!isNewNumberOfRepsValid) {
+        if (!isNewRepsValueValid) {
+            logger.log(Level.WARNING, "Repetition value is invalid.");
             throw new InvalidWorkoutException(className, InvalidWorkoutException.INVALID_REPS_VALUE_ERROR_MSG);
         }
 
         Workout updatedWorkout = workoutsList.get(indexToUpdate - 1);
-        updatedWorkout.setRepetitions(newNumberOfReps);
+        String exerciseName = updatedWorkout.getExerciseName();
+        boolean isWorkoutExist = checkForExistingWorkout(exerciseName, newRepsValue);
```
Comment (Musfirahe0556596) [inline src/main/java/data/workouts/WorkoutList.java:246]: Perhaps variable name could be "isExistingWorkout"?

**Review by Musfirahe0556596 — APPROVED** [review]
Looks good! Some changes to consider!

File: src/main/java/data/workouts/WorkoutList.java (hunk 2 above)
Comment (TianaiYan) [reply src/main/java/data/workouts/WorkoutList.java:246]: Sure. This sounds better.

**Review by a1021492980 — APPROVED** [review]

**Review by alanlowzies — APPROVED** [review]

**Pull request merged**